### PR TITLE
Common SESSION Bus

### PR DESCRIPTION
**Pull request by JasonLG1979**
This goes a long way to making plugins that can be loaded async, async.

### CODE REVIEW

File: pithos/plugin.py [hunk 1]
```diff
@@ -95,29 +96,46 @@ def _maybe_migrate_setting(new_setting, name):
...
-            plugin.settings = settings.get_child(settings_name)
-            _maybe_migrate_setting(plugin.settings, name)
-        else:
-            # Out of tree plugin
-            plugin.settings = Gio.Settings.new_with_path('io.github.Pithos.plugin',
+    def on_got_bus(source, result, userdata):
+        try:
+            bus = Gio.bus_get_finish(result)
+            logging.info('Got session bus')
+        except Glib.Error as e:
```
Comment (TingPing) [inline pithos/plugin.py]: __[flake8]__
* [F821] undefined name 'Glib'

File: pithos/plugin.py [hunk 2]
```diff
@@ -16,17 +16,21 @@
 import logging
 import glob
 import os
-from gi.repository import Gio
+from gi.repository import (
```
Comment (TingPing) [inline pithos/plugin.py]: __[flake8]__
* [F401] 'gi.repository.GLib' imported but unused

File: pithos/plugin.py [hunk 3]
```diff
@@ -95,29 +99,46 @@ def _maybe_migrate_setting(new_setting, name):
...
-            plugin.settings = settings.get_child(settings_name)
-            _maybe_migrate_setting(plugin.settings, name)
-        else:
-            # Out of tree plugin
-            plugin.settings = Gio.Settings.new_with_path('io.github.Pithos.plugin',
+    def on_got_bus(source, result, userdata):
+        try:
+            bus = Gio.bus_get_finish(result)
+            logging.info('Got session bus')
+        except Glib.Error as e:
```
Comment (TingPing) [inline pithos/plugin.py]: __[flake8]__
* [F821] undefined name 'Glib'

**Comment by TingPing** [comment]
Screensaver should handle in prepare not enable. Notify doesn't use this?

Also better commit titles man: `plugin: Share session bus between plugins`

**Comment by TingPing** [comment]
Mediakeys could also better handle an error message in the case that keybinder and dbus don't work.

**Comment by JasonLG1979** [comment]
> Screensaver should handle in prepare not enable.

OK.

> Notify doesn't use this?

It does but not in that way. It gets the session bus when it inits. For GioNotify it'll be a little more tricky we would need to handle the error in a callback.

> Mediakeys could also better handle an error message in the case that keybinder and dbus don't work.

Ok.

**Comment by JasonLG1979** [comment]
> Mediakeys could also better handle an error message in the case that keybinder and dbus don't work.

```python
return 'Keybinder requires X11'
```
and
```python
return 'Keybinder not found'
```

Could be changed to:

```python
return 'DBus binding failed and Keybinder requires X11'
```
and
```python
return 'DBus binding failed and Keybinder not found'
```

**Comment by JasonLG1979** [comment]
Something like that?

**Pull request merged**